### PR TITLE
Use local installation of tree-sitter

### LINKED ISSUE #2956: Eliminate sudo from the build
Various people are annoyed by the need for `sudo` to build semgrep. I know it's used to install libtree-sitter. I don't think it's used for anything else but we need to check.

* [Build for Conda](https://github.com/conda-forge/staged-recipes/pull/14494#issuecomment-820837740)
* [Build for Windows](https://github.com/returntocorp/semgrep/issues/1330)

### PR DESCRIPTION
**Pull request by mjambon**
Closes https://github.com/returntocorp/semgrep/issues/2956

test plan: all CI steps must succeed. In particular, the semgrep-core executable should be compiled successfully for all target platforms.

PR checklist:

- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] `changelog.d/<issue>.<type>` is a file with the _what_, _why_, and _how_ of the change.
  - \<issue> is `pa-312` (Linear ticket), `gh-1234` (GitHub issue), or `new-gizmo` (unique semantic name)
  - \<type> is `added`, `changed`, `fixed`, or `infra`.
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Changelog guidelines](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry)
- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by mjambon** [comment]
On alpine, we're getting a bunch of these errors that I'm not getting on ubuntu:
```
#33 [semgrep-core 19/21] RUN opam exec -- dune build
#33 13.30 File "src/ocaml-tree-sitter-core/src/bindings/lib/dune", line 12, characters 11-19:
#33 13.30 12 |     (names bindings)
#33 13.30                 ^^^^^^^^
#33 13.30 bindings.c:24:10: fatal error: tree_sitter/api.h: No such file or directory
#33 13.30    24 | #include <tree_sitter/api.h>
#33 13.30       |          ^~~~~~~~~~~~~~~~~~~
#33 13.30 compilation terminated.
#33 13.41 File "src/tree-sitter-lang/semgrep-bash/lib/dune", line 12, characters 18-26:
#33 13.41 12 |     (names parser bindings)
#33 13.41                        ^^^^^^^^
#33 13.41 bindings.c:6:10: fatal error: tree_sitter/api.h: No such file or directory
#33 13.41     6 | #include <tree_sitter/api.h>
#33 13.41       |          ^~~~~~~~~~~~~~~~~~~
#33 13.41 compilation terminated.
```
It makes sense since the angle brackets, as opposed to double quotes, are expected to search at the system locations and are not (?) supposed to consult the extra locations provided by `-I`. I need to figure out how to make this work everywhere and whether we should replace `<tree_sitter/api.h>` by `"tree_sitter/api.h"`.

Update: gcc's `-I` is supposed to extend the search path for both system and user search paths. The problem must be elsewhere, such as the wrong build command being issued to build semgrep-core.

**Comment by mjambon** [comment]
The upgrade of the tree-sitter runtime causes an apparent regression in one test in semgrep-rules. It was caused by invalid Java syntax in the target file, so I fixed that: https://github.com/returntocorp/semgrep-rules/pull/2319

File: cli/tests/e2e/snapshots/test_semgrep_core_parse_error/test_rule_parser__failure__error_messages/settings0/out.json [hunk 1]
```diff
@@ -19,12 +19,12 @@
         },
         {
           "end": {
-            "col": 12,
+            "col": 14,
```
Comment (aryx) [inline cli/tests/e2e/snapshots/test_semgrep_core_parse_error/test_rule_parser__failure__error_messages/settings0/out.json:22]: Why the column number changed here?

**Review by aryx — COMMENTED** [review]
Nice!
requiring sudo was ugly ...

File: cli/tests/e2e/snapshots/test_semgrep_core_parse_error/test_rule_parser__failure__error_messages/settings0/out.json (hunk 1 above)
Comment (kopecs) [reply cli/tests/e2e/snapshots/test_semgrep_core_parse_error/test_rule_parser__failure__error_messages/settings0/out.json:22]: This was from a change in tree-sitter error recovery. See also https://returntocorp.slack.com/archives/C01NXGX2EHZ/p1660624674796879

**Pull request merged**